### PR TITLE
Mfa merge

### PR DESCRIPTION
**Pull request by jakubvul**

### CODE REVIEW

File: source/auth_n.rst [hunk 1]
```diff
@@ -4915,24 +4875,33 @@ Challenging During Factor Creation
...
+      },
+      "challenge": {
+        "message": "Welcome to the Example! Your authorization code is ${code}"
+      }
     }
-  }
+
+.. only:: java
+
+  To create a challenge at the same time as you create the SMS Factor, you need to call ``account.createFactor((Factors.SMS.newCreateRequestFor(factor).createChallenge()).build())``:
```
Comment (mrioan) [inline source/auth_n.rst]: @mrafiei this looks weird. You have code in one line followed by a Java block showing more code. There seems to be some text missing to properly identify what that last code is for.

File: source/auth_n.rst [hunk 2]
```diff
@@ -5289,21 +5241,24 @@ Which will return:
...
+  You would then send a POST to the ``challenges`` collection which would generate a new Challenge and send an SMS message to the number specified in the Factor's Phone resource.
 
-  .. only:: csharp or vbnet
+.. only:: java
 
-    .. todo::
+  You would then challenge the factor which would generate a new Challenge and send an SMS message to the number specified in the Factor's Phone resource.
 
-      (dotnet.todo)
+   .. literalinclude:: code/java/authentication/mfa_challenge_existing_factor.java
```
Comment (mrioan) [inline source/auth_n.rst]: @mrafiei This code block is slightly intended to the right.

File: source/auth_n.rst [hunk 3]
```diff
@@ -4003,21 +3999,30 @@ First, you create the Account:
 Adding an SMS Factor
 ^^^^^^^^^^^^^^^^^^^^
 
-To add an additional SMS Factor to this Account, you send a POST to that Account's ``/factors`` endpoint:
+To add an additional SMS Factor to this Account, you send the following request:
 
-.. code-block:: http
+.. only:: rest or csharp or nodejs or php or python or vbnet
```
Comment (mrioan) [inline source/auth_n.rst:4004]: @mrafiei This code block is slightly intended to the right.

**Review by mrioan — COMMENTED** [review]
Additionally, @jakubswiatczak not sure if this is intended or not but `How API Key Authentication Works in Stormpath` in sidebar is not showing up in Rest (it does in Java)

**Comment by mrioan** [comment]
4 minor comments. You can consider this approved after they are fixed. Thanks!

File: source/auth_n.rst (hunk 1 above)
Comment (jakubvul) [reply source/auth_n.rst]: OK this actually isn't something that I can fix. @mrafiei any suggestions?

File: source/auth_n.rst (hunk 3 above)
Comment (jakubvul) [reply source/auth_n.rst:4004]: Resolved.

**Comment by jakubvul** [comment]
@mrioan API Key Auth is intentionally only for the SDK guides.

```
.. only:: not rest <--

  4.7. How API Key Authentication Works in Stormpath
  =====================================================
```

**Pull request merged**